redfish_client/resource.py

```diff
@@ -55,9 +54,9 @@ def _init_from_oid(self, oid):
             url, fragment = oid, ""
 
         resp = self._connector.get(url)
-        if resp.status_code != 200:
-            raise Exception(resp.content)
-        return dict(resp.headers), self._get_fragment(resp.json(), fragment)
+        if resp.status != 200:
+            raise Exception(resp.raw)
```

Comment:
Considering 

- this issue https://github.com/xlab-si/redfish-client-python/issues/8
- the fact we're IFing for status quite often

I'd propose to dedicate a real class to Response (not just named tuple) and equip it with utility methods for http status checks. I've seen something simular in requests library the other day and I liked the ease of use:

```python
class Response(...):
  def raise_for_auth():
    if self.status == 401:
      raise AuthException()

  def raise_for_invalid_response():
    if not self.json.is_valid:  # I made that up, but you get the point
      raise InvalidResponseException()

  def raise_for_error():
    self.raise_for_auth()
    self.raise_for_invalid_response()
```

Reply:
I have nothing against creating a more substantial class for Response, but at the moment when I was refactoring the connector, there was no urgent need for such a thing.

What I wanted to create is a uniform API that library users and higher levels of Redfish client library can use in any way they want. One such use case that we have right now is proxying requests to the Redfish API where we want to return most of the response data unaltered back to the client. Catching exceptions in scenarios where we expect a certain fraction of requests to return with non-OK response status makes little sense.

Reply:
I feel your aggregator-oriented point of view. Better exception handling would be nice for Ansible modules (and any end-user in general, I think) but the way you set it now, it's easily upgradable with followup PR if we decide to do so, so 👍